list-clusters

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+set -Eeuo pipefail
+
+unrecognised_command() {
+	echo "Unrecognised command: $*"
+	exit 1
+}
+
+print_help() {
+	echo 'https://github.com/shiftstack/shiftstack-ci'
+	echo
+	echo 'list-clusters [ -a | -s ] [ -l ]'
+	echo
+	echo 'Prints the IDs of the detected clusters, based on their Network.'
+	echo
+	echo -e '\t-a only lists active clusters'
+	echo -e '\t-s only lists stale clusters'
+	echo
+	echo -e '\t-l prints the full cluster name. Otherwise, truncates at 14 characters'
+	echo
+	echo 'Clusters are identified as stale if their network is more than 5 hours old.'
+}
+
+print_cluster_id() {
+	declare \
+		cluster_id="$1" \
+		format="$2"
+
+	case "$format" in
+		'long' ) echo "$cluster_id" ;;
+		'short') echo "${cluster_id:0:14}" ;;
+		*  ) >&2 echo "wrong format '$format'" ; exit 1 ;;
+	esac
+}
+
+VALID_LIMIT="$(date --date='-5 hours' +%s)"
+readonly VALID_LIMIT
+
+declare filter=''
+declare format='short'
+
+while getopts lash o; do
+	case "$o" in
+		l) format='long'             ;;
+		a) filter='active'           ;;
+		s) filter='stale'            ;;
+		h) print_help; exit          ;;
+		*) unrecognised_command "$@" ;;
+	esac
+done
+
+for network in $(openstack network list -c Name -f value); do
+	if [[ $network = *-*-openshift ]]; then
+		CLUSTER_ID="${network/-openshift/}"
+		case "$filter" in
+			'active')
+				CREATION_TIME=$(openstack network show "$network" -c created_at -f value)
+				CREATION_TIMESTAMP=$(date --date="$CREATION_TIME" +%s)
+				if ! [[ $CREATION_TIMESTAMP < $VALID_LIMIT ]]; then
```

Comment:
Took me a while to notice the `!` here. Suggest you do:

```
if [[ $CREATION_TIMESTAMP > $VALID_LIMIT ]];
```

We don't really care about the case where $CREATION_TIMESTAMP == $VALID_LIMIT.

Reply:
but... the resolution is in _seconds_! :scream:

Reply:
Since those are numbers anyway, I have changed the comparison operators to `-ge` and `-lt`. WDYT?

cf https://www.tldp.org/LDP/abs/html/comparison-ops.html